pylhc/machine_settings_info.py

```diff
@@ -394,19 +416,24 @@ def write_trim_histories(
 # Beamprocess ##################################################################
 
 
-def _get_beamprocess(acc_time: AccDatetime, accel: str, source: str) -> DotDict:
-    """Get the info about the active beamprocess at ``acc_time``."""
-    fill_no, fill_bps = LSA.find_last_fill(acc_time, accel, source)
-    beamprocess = LSA.find_active_beamprocess_at_time(acc_time)
+def _get_beamprocess(acc_time: AccDatetime, accel: str, source: str, beamprocess: str = None) -> DotDict:
+    """Get the info about the active beamprocess at ``acc_time`` or the given one."""
+    if beamprocess is None:
+        beamprocess = LSA.find_active_beamprocess_at_time(acc_time, accelerator=accel)
+    bp_info = LSA.get_beamprocess_info(beamprocess)
+    bp_info = DotDict(bp_info)
+
+    fill_no, fill_bps = LSA.find_last_fill(acc_time, accelerator=accel, source=source)
+    bp_info.Fill = fill_no
+
     try:
-        start_time = _get_beamprocess_start(fill_bps, acc_time, str(beamprocess))
+        start_time = _get_beamprocess_start(fill_bps, acc_time, bp_info.Name)
     except ValueError as e:
         raise ValueError(f"In fill {fill_no} the {str(e)}") from e
-    bp_info = LSA.get_beamprocess_info(beamprocess)
-    bp_info.update({"Fill": fill_no, "StartTime": start_time})
+    bp_info.StartTime = start_time
```

Comment:
Should this line be in the `try` block? It seems to me like `AccDatetime` initiation will not return a value if an error occurs.

Reply:
No, the error is raised anyway. Only excepted to add info